### PR TITLE
Added user account endpoint to update info

### LINKED ISSUE #42: Account control panel for updating user's emails and passwords

### PR DESCRIPTION
**Pull request by Ommy**
Resolves #42

### CODE REVIEW

**Comment by Kapin** [comment]
Main question is whether this belongs in api or in auth

**Comment by Trinovantes** [comment]
I'd say api

File: src-django/api/views.py [hunk 1]
```diff
@@ -37,6 +38,34 @@ def generate(self, request, pk=None):
...
+    model = User
+    queryset = User.objects.all()
+    serializer_class = serializer.UserSerializer
+    allowed_changes = ['first_name', 'last_name', 'email', 'password']
+
+    @detail_route(methods=['patch'])
+    def update_details(self, request, pk=None):
+        if pk is None or not request.body:
+            return HttpResponseBadRequest()
+        user = User.objects.get()
```
Comment (Kapin) [inline src-django/api/views.py]: this should specify the primary key, correct?

File: src-django/api/urls.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 router.register(r'elements', views.ElementViewSet, base_name='element')
 router.register(r'concepts', views.ConceptViewSet, base_name='concept')
 router.register(r'conditionals', views.ShowIfViewSet, base_name='conditional')
+router.register(r'user', views.UserViewSet, base_name='user')
```
Comment (Kapin) [inline src-django/api/urls.py]: users?

**Comment by Ommy** [comment]
screen shot 2016-02-18 at 4 44 43 pm

**Comment by Ommy** [comment]
Let me know how this should look, this was just an initial look.

File: src-backbone/app/js/models/session.js [hunk 3]
```diff
@@ -96,6 +97,43 @@ let SessionModel = Backbone.Model.extend({
...
+            beforeSend: function() {
+                App().RootView.showSpinner();
+            },
+            complete: function() {
+                App().RootView.hideSpinner();
+            },
+            success: function(response) {
+                App().RootView.clearNotifications();
+                App().RootView.showNotification("Successfuly updated account details!");
+                Helpers.navigateToDefaultLoggedIn();
```
Comment (Trinovantes) [inline src-backbone/app/js/models/session.js]: This isn't necessary

**Comment by Trinovantes** [comment]
Looks fine. I'll update the UI later if there's time after conditional logic

**Comment by Trinovantes** [comment]
This also closes #238

File: src-backbone/app/js/models/session.js [hunk 4]
```diff
@@ -96,6 +97,46 @@ let SessionModel = Backbone.Model.extend({
...
+                App().RootView.showSpinner();
+            },
+            complete: function() {
+                App().RootView.hideSpinner();
+            },
+            success: function(response) {
+                App().RootView.clearNotifications();
+                App().RootView.showNotification({
+                    title: 'Success!',
+                    desc: "Successfuly updated account details!",
```
Comment (Trinovantes) [inline src-backbone/app/js/models/session.js]: add this string to the locales files

File: src-backbone/app/js/models/session.js [hunk 5]
```diff
@@ -96,6 +97,46 @@ let SessionModel = Backbone.Model.extend({
...
+                    alertType: 'success',
+                });
+            },
+            error: function(errors) {
+                App().RootView.clearNotifications();
+                let parsedErrors = JSON.parse(errors.responseText)
+                Object.keys(parsedErrors).forEach(function(key) {
+                    App().RootView.showNotification({
+                        title: 'There was a problem',
```
Comment (Trinovantes) [inline src-backbone/app/js/models/session.js]: this string too

File: src-backbone/app/js/models/session.js (hunk 4 above)
Comment (Ommy) [inline src-backbone/app/js/models/session.js]: Done

File: src-backbone/app/js/models/session.js (hunk 5 above)
Comment (Ommy) [inline src-backbone/app/js/models/session.js]: Done

File: src-backbone/app/js/models/session.js [hunk 6]
```diff
@@ -96,6 +97,46 @@ let SessionModel = Backbone.Model.extend({
         });
     },
 
+    updateInformation: function(formData) {
+        let self = this;
+        let json = {};
+        formData.forEach(function(item) {
+            if (item.value !== "") {
```
Comment (connorcimowsky) [inline src-backbone/app/js/models/session.js]: possibly silly question: will this allow `null`? I think `if (item.value)` will be truthy for non-`null`, non-empty strings.

**Comment by connorcimowsky** [comment]
nice! LGTM

File: src-backbone/app/js/models/session.js (hunk 6 above)
Comment (Ommy) [inline src-backbone/app/js/models/session.js]: The form data itself will always give us an empty string for a field if it hasn't been filled in.

**Pull request merged**